src/loose/state.js

```diff
@@ -158,4 +158,10 @@ export class LooseParser {
       plugin(this, pluginConfigs[name])
     }
   }
+
+  parse() {
+    let node = this.options.program || this.startNode()
+    this.next()
```

Comment:
Yes, the loose parse class has a `next` method, which wraps the regular tokenizer with some error-correcting magic.

I think you should just move the body from `parse_dammit` into this method (which I guess you did) and have `parse_dammit` call it, to avoid duplication.